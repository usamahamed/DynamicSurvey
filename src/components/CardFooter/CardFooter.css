.card-footer {
  padding: 20px 0 0;
}

.card-footer .button:first-child {
  margin-right: 10px;
}
