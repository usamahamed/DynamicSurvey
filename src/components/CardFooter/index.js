import React from 'react';
import PropTypes from 'prop-types';

import Button from '../Button';
import './CardFooter.css';

const CardFooter = props => (
  <footer className="clearfix card-footer">
    <div className="right">
      <Button buttonClick={props.clickBack} className="button--flat">
        BACK
      </Button>
      <Button
        fieldValue={props.fieldValue}
        buttonClick={props.clickSubmit}
        className="button--success"
      >
        SUBMIT
      </Button>
    </div>
  </footer>
);

CardFooter.defaultProps = {
  fieldValue: '',
};

CardFooter.propTypes = {
  fieldValue: PropTypes.string,
  clickBack: PropTypes.func.isRequired,
  clickSubmit: PropTypes.func.isRequired,
};

export default CardFooter;
