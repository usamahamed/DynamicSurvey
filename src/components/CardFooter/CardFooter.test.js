import React from 'react';
import { shallow } from 'enzyme';
import CardFooter from './index';

describe('<CardFooter/>', () => {
  const clickBack = jest.fn();
  const clickSubmit = jest.fn();
  const wrapper = shallow(<CardFooter clickBack={clickBack} clickSubmit={clickSubmit} />);

  it('renders the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders two buttons', () => {
    expect(wrapper.find('Button').length).toEqual(2);
  });
});
