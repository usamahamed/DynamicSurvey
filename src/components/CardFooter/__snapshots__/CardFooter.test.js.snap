// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardFooter/> renders the component 1`] = `
<footer
  className="clearfix card-footer"
>
  <div
    className="right"
  >
    <Button
      buttonClick={[Function]}
      className="button--flat"
      fieldValue={null}
    >
      BACK
    </Button>
    <Button
      buttonClick={[Function]}
      className="button--success"
      fieldValue=""
    >
      SUBMIT
    </Button>
  </div>
</footer>
`;
